File: Codeforces/_Problems/DominoPiling.py
width = int(input())
length = int(input())
ans = 0

def q1(width,length):
    if width%2 == 0:
        ans = width/2 * length
        return ans

    if(length%2==0):
        ans = length/2 * width
        return ans

    else:
        return q1(width,length-1)+(width-1)/2

print(int(q1(width,length)))

File: Codeforces/_Problems/DominoPiling.cpp
#include <iostream>

using namespace std;

int q1(int w, int l){
    int ans;

    if(w%2 ==0){
        return w/2 * l;
    }
    else if (l%2== 0 ){
        return l/2 * w;
    }
    else{
        return q1(w,l-1)+(w-1)/2;
    }
}

int main (){
    int w,l;
    int ans;
    cin>>w;
    cin>>l;

    ans=q1(w, l);
    cout<<ans;
    return 0;
}


